connectors/utils.py

```diff
@@ -352,3 +352,16 @@ def get_event_loop(uvloop=False):
             loop = asyncio.new_event_loop()
             asyncio.set_event_loop(loop)
     return loop
+
+
+def nested_get(dictionary, keys, default=None):
```

Comment:
I'm kinda amazed that this isn't part of the language already. Nice.

Reply:
that's not something I've ever need, and I suspect you don't.

we would usually do:

```
self.features["sync_rules"]["basic"]["enabled"]
```

now if you want to have defaults and have that line works even for empty dicts, you would use `collections.defaultdict` https://docs.python.org/3/library/collections.html#collections.defaultdict and implement the behavior you want for defaults.

Reply:
But this wouldn't work for arbitrary levels right?
```
>>> from collections import defaultdict
>>> features = defaultdict(lambda: defaultdict(lambda: defaultdict(lambda: 'default')))
>>> features['a']['b']['c']['d']
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: string indices must be integers
```

Reply:
As agreed with @tarekziade in a 1:1 discussion we'll move nested_get to keep it locally to the `Features` class.